### PR TITLE
filamesh: remove duplicated structs, add unit test.

### PR DESCRIPTION
**Pull request by prideout**
- Remove duplicated Box class
- Add filamesh header and remove duplicated filamesh Header and Part structs.
- Rename `interleaved` to `flags`.
- Add unit test.

This prepares for a couple upcoming features (snorm16 texture coordinates and mesh compression).

### CODE REVIEW

File: libs/filameshio/include/filameshio/filamesh.h [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+
+static const uint32_t VERSION = 1;
+
+enum IndexType : uint32_t {
+    UI32 = 0,
+    UI16 = 1,
+};
+
+enum Flags : uint32_t {
+    INTERLEAVED         = (1 << 0),
```
Comment (romainguy) [inline libs/filameshio/include/filameshio/filamesh.h]: The parenthesis are not needed there

File: tools/filamesh/README.md [hunk 2]
```diff
@@ -29,7 +29,7 @@ Note: the UV1 attribute cannot be used in interleaved mode
     uint32  : number of parts (sub-meshes or draw calls)
     float3  : center of the total bounding box (AABB)
     float3  : half extent of the total bounding box (AABB)
-    uint32  : 0 if the vertices are not interleaved, 1 otherwise
+    uint32  : flags (interleaved, UV type, compression type)
```
Comment (romainguy) [inline tools/filamesh/README.md]: Document the flags

File: libs/filameshio/include/filameshio/filamesh.h [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#ifndef TNT_FILAMENT_FILAMESHIO_FILAMESH_H
+#define TNT_FILAMENT_FILAMESHIO_FILAMESH_H
+
+#include <filament/Box.h>
+
+namespace filamesh {
```
Comment (romainguy) [inline libs/filameshio/include/filameshio/filamesh.h]: Is this header only for testing? We shouldn't need to make this public

Comment (prideout) [reply libs/filameshio/include/filameshio/filamesh.h]: The header is used by `libs/filameshio` (the reader) and `tools/filamesh` (the writer), and `test_filamesh`.  It does not need be installed so it is not included in DIST_HDRS.

File: tools/filamesh/README.md (hunk 2 above)
Comment (prideout) [reply tools/filamesh/README.md]: Done

File: libs/filameshio/include/filameshio/filamesh.h (hunk 1 above)
Comment (prideout) [reply libs/filameshio/include/filameshio/filamesh.h]: Fixed

File: libs/filameshio/include/filameshio/filamesh.h (hunk 3 above)
Comment (romainguy) [reply libs/filameshio/include/filameshio/filamesh.h]: Makes sense, thanks

**Pull request merged**